room_allocation.py

```diff
@@ -104,22 +120,246 @@ def get_room_by_name(self, room_name):
+
+    def print_room(self, room_name):
+        '''
+        Prints names of people in specified room
+        '''
+        if room_name not in self.rooms:
+            return "Room does not exist."
+        people = self.get_room_by_name(room_name).people_in_room
+
+        if len(people.keys()) > 0:
+            return people.keys()
+        return "No people in %s." % (room_name)
+
+    def is_valid_filename(self, filename):
+        for char in filename:
+            if char in "\/:*?<>|":
+                return False
+        return True
+
+    def print_unallocated(self, filename=None):
+        '''
+        Print list of people not in a room
+        '''
+        if len(self.unallocated_people) < 1:
+            return "No unallocated people exist."
+
+        all_data = '\n'.join(self.unallocated_people.keys())
+
+        if filename is None:
+            return all_data
+
+        directory = "test_files/"
+        filepath = directory + filename + '.txt'
+        if filename and self.is_valid_filename(filename):
+            f = open(filepath, "w")
+            f.write(all_data)
+            f.close()
+            return "Wrote to " + filepath + " successfully."
+        else:
+            return "Invalid filename."
+
+    def save_state(self, db='amity'):
+        '''
+        Persists the data of the app in database and stores it in the 
+        /databases folder
+        args:
+            db - sqlite database to save to (name to give databases)
+
+        '''
+
+        all_rows = []  # both people and room rows
+        for room_obj in self.rooms.values():
+            row = room_obj.to_db_row()
+            row.people_in_room = room_obj.people_in_room_to_rows()
+            all_rows.append(row)
+            all_rows += row.people_in_room
+
+        for person_obj in self.unallocated_people.values():
+            row = person_obj.to_db_row()
+            all_rows.append(row)
+
+        directory = "databases/"
+        filepath = directory + db + '.db'
+        if self.is_valid_filename(db):
+            with open(filepath, 'wb') as f:
+                pass
+            db = connect_db(filepath)
+            db.add_all(all_rows)
+            db.commit()
+            return "Data was successfuly saved to " + filepath
+
+    def load_state(self, dbpath):
+        '''
+        Loads data from a database into Amity.
+        args:
+            dbpath - sqlite database to load from
+        '''
+        if not os.path.exists(dbpath):
+            return "Invalid file path."
+
+        db = connect_db(dbpath)
+        rooms = db.query(ModelRoom).all()
+        people = db.query(ModelPerson).all()
+
+        for room in rooms:
+            if not room.name in self.rooms:
+                self.rooms[room.name] = room.to_obj()
+                people_in_room = room.people_in_room_to_objs()
+                self.rooms[room.name].people_in_room = people_in_room
+            else:
+                return "Room already exists."
```

Comment:
This return might be premature. A `continue` statement is preferable. If you had 20 rooms and the first room is already in the list of rooms, it means it won't check the rest of the rooms. But a continue will still check and add the rest of the rooms.